cuML/src/glm/qn/qn_solvers.h

```diff
@@ -275,8 +277,9 @@ inline OPT_RETCODE min_owlqn(const LBFGSParam<T> &param, Function &f,
       return OPT_NUMERIC_ERROR;
     }
     // Save the curent x and gradient
-    xp = x;
-    gradp = grad;
+    xp.copy_async(x, stream);
+    gradp.copy_async(grad, stream);
+    CUDA_CHECK(cudaStreamSynchronize(stream));
```

Comment:
See my comment on line 143 is this `cudaStreamSynchronize` necessary?

Reply:
See above